agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -67,36 +83,136 @@
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @AutoService(AutoConfigurationCustomizerProvider.class)
-public class OpenTelemetryConfigurer implements AutoConfigurationCustomizerProvider {
+public class SecondEntryPoint implements AutoConfigurationCustomizerProvider {
+
+  private static final Logger startupLogger =
+      LoggerFactory.getLogger("com.microsoft.applicationinsights.agent");
 
-  @Nullable public static LoggerExporter loggerExporter;
+  @Nullable public static AgentLogExporter agentLogExporter;
 
   @Nullable private static BatchLogProcessor batchLogProcessor;
   @Nullable private static BatchSpanProcessor batchSpanProcessor;
   @Nullable private static MetricReader metricReader;
 
   @Override
   public void customize(AutoConfigurationCustomizer autoConfiguration) {
-    TelemetryClient telemetryClient = TelemetryClient.getActive();
-    if (telemetryClient == null) {
-      // agent failed during startup
-      return;
+
+    File tempDir =
+        TempDirs.getApplicationInsightsTempDir(
+            startupLogger,
+            "Telemetry will not be stored to disk and retried later"
+                + " on sporadic network failures");
+
+    Configuration config = FirstEntryPoint.getConfiguration();
```

Comment:
after parsing the json config file, can we make some of the properties available for standalone exporter? It's currently tied to TelemetryClient. 
globalTags and globalProperties are applicable to all entity's customDimension

Reply:
standalone exporter doesn't use json config file

Reply:
oh, how does cx configure it? do we offer any customization?

Reply:
programmatic configuration only at this point, e.g. they could write their own span processor if they want to add global tags/properties